### PR TITLE
[1/2] clean-up / general improvements

### PR DESCRIPTION
**Pull request by onur-ozkan**
This PR applies various clippy suggestions on the tools. I have only applied the ones that make sense and left out trivial changes (e.g., suggestions like 'remove &' are ignored to keep the original commit history for the lines).

I am planning to do the same for the library and compiler, but those will add too many changes to this PR, so I will handle them in a separate PR later.

### CODE REVIEW

File: src/tools/compiletest/src/runtest/debugger.rs [hunk 1]
```diff
@@ -148,5 +148,5 @@ fn check_single_line(line: &str, check_line: &str) -> bool {
         rest = &rest[pos + current_fragment.len()..];
     }
 
-    if !can_end_anywhere && !rest.is_empty() { false } else { true }
+    !(!can_end_anywhere && !rest.is_empty())
```
Comment (jieyouxu) [inline src/tools/compiletest/src/runtest/debugger.rs:151]: Suggestion: this is equivalent to `can_end_anywhere || rest.is_empty()`, I think?

**Review by jieyouxu — APPROVED** [review]
compiletest changes LGTM, just one nit

File: src/tools/lint-docs/src/lib.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl Lint {
         for &expected in &["### Example", "### Explanation", "{{produces}}"] {
             if expected == "{{produces}}" && self.is_ignored() {
                 if self.doc_contains("{{produces}}") {
-                    return Err(format!(
+                    return Err(
                         "the lint example has `ignore`, but also contains the {{{{produces}}}} marker\n\
```
Comment (kadiwa4) [inline src/tools/lint-docs/src/lib.rs:88]: it seems you forgot to unescape the braces in this string

**Review by albertlarsan68 — COMMENTED** [review]
LGTM for the general changes, but let's wait on the maintainers of the tools

File: src/tools/lint-docs/src/lib.rs (hunk 2 above)
Comment (onur-ozkan) [reply src/tools/lint-docs/src/lib.rs]: Done https://github.com/rust-lang/rust/compare/a7ab37d0d510da9d9f4a4a93f31aef10c8f277e3..eaa45c8b28468e8ec46bf364eeead996bcc3df8c, thanks!

**Comment by Kobzol** [comment]
`opt-dist` change fine by me :+1:

**Comment by albertlarsan68** [comment]
I think everyone approved, so @bors r+

**Comment by onur-ozkan** [comment]
Rebased.

@bors r=albertlarsan68

**Pull request merged**